enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,412 @@
+   passing a single test name to run as the argument, and reports back the result
+   to the main process.
+6. Present the summary results of running tests.
+
+Below diagram shows the current flow:
+
+![Old Flow](./old-flow.png)
+<!--
+Above diagram was generated at https://mermaid.live/ using following code:
+sequenceDiagram
+    actor U as User
+    participant O as openshift-tests
+    U->>+O: run openshift/conformance
+    O->>+O: list tests
+    loop over all tests
+        O->>+O: run #lt;test name#gt;
+        O->>+U: results
+    end
+-->
+
+### Workflow Description - new
+
+This proposal's goal is to extend test discovery (step 1), and running the actual
+test (step 5) with external binaries, which are retrieved from the payload image.
+The improved flow would be as follows:
+1. Retrieve test binaries from the payload image.
+2. Discover all tests, both the built-in into `openshift-tests` binary, and the
+   ones provided by external binaries by executing `<binary>-tests list`.
+   Split them into suites based on test names/labels.
+3. Filter tests based on: desired suite, configuration/environment of the cluster
+   being tested, parallel/serial requirements.
+4. Discover the deployment topology and set up all the arguments necessary
+   to monitor the cluster and run the tests.
+5. Start monitoring the cluster functionality.
+6. Iterate overall all the test creating a subprocess invoking `<binary>-tests run-test`
+   passing a single test to run as the argument, and report back the results to the
+   main process.
+7. Present the summary results of running tests.
+
+Below diagram shows the proposed flow:
+
+![New Flow](./new-flow.png)
+<!--
+Above diagram was generated at https://mermaid.live/ using following code:
+sequenceDiagram
+    actor U as User
+    participant O as openshift-tests
+    participant E as external-tests
+    participant R as release:tag
+    U->>+O: run openshift/conformance
+    O->>+O: list tests
+    O->>+R: extract test binary
+    rect rgb(177, 177, 177)
+        opt test binary exists?
+            R->>+E: external-tests
+            O->>+E: list tests
+            E->>+O: tests
+        end
+    end
+    loop over all tests
+        alt embedded test?
+            O->>+O: run #lt;test name#gt;#10;
+        else external test?
+            rect rgb(177, 177, 177)
+                O->>+E: run #lt;test name#gt;
+                E->>+O: results
+            end
+        end
+    end
+    O->>+U: results
+-->
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+To achieve the first step of the proposal, we need to:
+1. Provide a library for building `<binary>-tests` which exposes two commands:
+   - `list` - responsible for listing tests in JSON format;
+   - `run` - run a single test, returning results in ginkgo compatible format;
+2. Provide an image annotation or similar mechanism informing where the test
+   binary is embedded in the release payload.
```

Comment:
the library also needs to help teams:
1) indicate+maintain labels/tags that should be dynamically added to their tests (rules.go/annotations.go)
2) manage the use of disconnected images (i.e. ensure they are only referencing images that are mirrored, at least for conformance tests)

Reply:
Done.

Reply:
not seeing where this got added (maybe you haven't pushed yet)